### PR TITLE
Add permission detail to docs

### PR DESCRIPTION
**Pull request by mknowlton89**
### Features and Changes

This PR introduces a new article within our docs that outlines how our roles and permissions work. The link to this can be found in the first comment below.

There wasn't a great section to add this  within our current docs, so I added a new section `Account` and nested this as the first article within that section.

Screen Shot 2023-08-24 at 2 10 12 PM

### CODE REVIEW

File: docs/docs/account/user-permissions.mdx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+have the added ability to give a user project-level roles.
+
+:::note
+
+For instance, you can set a user's global role as `Collaborator`, which allows them to view features and experiements, as well as add
+comments and contribute ideas, and then give them an `Experimenter` role for a particular project, which allows them to create and run experiments, but only for the specified project.
+
+:::
+
+Below are the roles currently available in GrowthBook, ordered from least permissive to most. Each additional level inherits all of the permissions from the roles below it.
```
Comment (lukesonnet) [inline docs/docs/account/user-permissions.mdx:20]: >ordered from least permissive to most.

This isn't correct. `engineer` and `analyst` are non-overlapping permissions.

> Each additional level inherits all of the permissions from the roles below it.

This is confusing. Maybe you mean "above" it?

File: docs/docs/account/user-permissions.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+- `readonly` - This role allows the user to view feature flags and experiments, but not make any changes.
+- `collaborator` - This role allows the user to view feature flags and experiments, as well as add comments and contribute ideas.
+- `engineer` - This role allows the user to manage feature flags, but not experiments.
+- `analyst` - This allows the user to analyze experiments.
+- `experimenter` - This allows the user to create feature flags and run experiments.
+- `admin` - This allows the user to manage all aspects of GrowthBook, including billing, organizational settings, and team members.
+
+And finally, there is one last level to GrowthBook's user permissions, which determines environment specific limits. This permission level only applies to `engineer` and `experimenter` roles in organizations in a `Pro` or `Enterprise` account. This
```
Comment (lukesonnet) [inline docs/docs/account/user-permissions.mdx:29]: This section could use a header or more clear context, maybe a relevant screenshot?

**Review by lukesonnet — CHANGES_REQUESTED** [review]
Thanks for doing this, much needed.

1. Please see inline for some issues with clarity. 
2. Overall, If feel like we might want to also have a mapping from some core actions and the roles that can do it:

* Create and edit Metrics -> ...
* Create and edit Feature Flags -> ...
* Create and edit Experiments -> ....

File: docs/docs/account/user-permissions.mdx (hunk 1 above)
Comment (mknowlton89) [reply docs/docs/account/user-permissions.mdx]: I just took this out - it was confusing either way.

File: docs/docs/account/user-permissions.mdx (hunk 2 above)
Comment (mknowlton89) [reply docs/docs/account/user-permissions.mdx:33]: Updated!

File: docs/docs/account/user-permissions.mdx (hunk 1 above)
Comment (mknowlton89) [reply docs/docs/account/user-permissions.mdx]: I also updated the descriptions of each role, showing what role it inherits permission from explicitly.

Taking a pass at adding a table that has more granular view/edit permissino
s

Comment (jdorn) [reply docs/docs/account/user-permissions.mdx]: The way I've seen this elsewhere, the axes would be swapped.  One row per feature and the columns are the different roles.

Comment (mknowlton89) [reply docs/docs/account/user-permissions.mdx]: Just pushed a commit that introduces the table with the axes as @jdorn described.

I'd love feedback on the rows, along with the descriptors for the various permission levels.

**Review by tzjames — APPROVED** [review]

File: docs/docs/account/user-permissions.mdx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+:::note
+
+For instance, you can set a user's global role as `Collaborator`, which allows them to view features and experiements, as well as add
+comments and contribute ideas, and then give them an `Experimenter` role for a particular project, which allows them to create and run experiments, but only for the specified project.
+
+:::
+
+Below are the roles currently available in GrowthBook and their associated permissions.
+
```
Comment (tzjames) [inline docs/docs/account/user-permissions.mdx:21]: Also for docs changes I often run them by Chat-GTP asking them to rephrase for clarity.  For the most part I like its suggestions slightly better than what you have:

GrowthBook provides user roles and permissions at both the global and project level. This feature enables you to manage who can access GrowthBook and their level of access.

When inviting a new user to your organization, you can assign a global role, which defines their permissions across all projects. If you have a `Pro` or `Enterprise` account, you can also assign project-level roles to users.

:::note
For example, you can assign a user the global role of `Collaborator`, allowing them to view features and experiments, add comments, and contribute ideas. You can then assign them an `Experimenter` role for a specific project, which allows them to create and run experiments, but only for that project.
:::

The table below lists the roles available in GrowthBook and their associated permissions.

File: docs/docs/account/user-permissions.mdx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+|    Manage Plan    |        No         |              No               |                      No                       |                          No                           |                          No                           |                          Yes                          |
+|  Manage Billing   |        No         |              No               |                      No                       |                          No                           |                          No                           |                          Yes                          |
+
+#### Environment Specific Limits
+
+![Environment Specific Limits](/images/user-permissions-env-specific.png)
+
+And finally, there is one last level to GrowthBook's user permissions, which determines environment specific limits. This permission level only applies to `engineer` and `experimenter` roles in organizations in a `Pro` or `Enterprise` account. This
+allows you to limit the feature flags and experiments a user can manage to particular environments. For instance, you can allow and engineer to create and run experiments in a `staging` environment, but not in `production`.
```
[the remark this answers is not in the document]
Comment (mknowlton89) [reply docs/docs/account/user-permissions.mdx:48]: I think `and` is correct - only `engineer` and `experimenter` GB roles can have environment restrictions. `readonly`, `collaborator`, and `analyst` roles can't make changes to feature flags, so giving them environment specific restrictions doesn't actually change what they can/can't do, and `admins` don't have any restrictions.

File: docs/docs/account/user-permissions.mdx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+#### How permissions are evaluated
+
+When evaluating whether or not a user has permissions to perform a certain action, GrowthBook will check if the action is taking place within a particular project, and if so, we'll check their project-level permissions first, before falling back to
+the user's global role. If the action is not taking place within a project, we'll just check the user's global role.
+
+If your organization is not enrolled in a `Pro` or `Enterprise` account, we will simply evaluate the user's permission level via their global role.
```
Comment (tzjames) [inline docs/docs/account/user-permissions.mdx:55]: And some more chat-gpt inspired rephrasings that I think are slightly more clear here:

GrowthBook's user permissions also include environment specific limits. This permission level applies only to `engineer` and `experimenter` roles in `Pro` or `Enterprise` accounts. It allows you to limit the feature flags and experiments a user can manage to specific environments. For example, you can allow an engineer to create and run experiments in a `staging` environment, but not in `production`.

#### How permissions are evaluated

GrowthBook evaluates user permissions based on whether an action is taking place within a specific project. If so, project-level permissions are checked first, followed by the user's global role. If the action is not within a project, only the user's global role is checked.

For organizations without a `Pro` or `Enterprise` account, user permissions are evaluated solely based on their global role.

File: docs/docs/account/user-permissions.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (tzjames) [reply docs/docs/account/user-permissions.mdx:48]: ah sorry the second comment is confusing you.  I was talking about the last sentence: 

> For instance, you can allow and engineer to create and run experiments in a `staging` environment, but not in `production`
It is the `and` there that is gramatically incorrect.  The second comment was suggesting perhaps to put engineer in backticks.

File: docs/docs/account/user-permissions.mdx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+|   Environments    |       View        |             View              |         View<br></br>Add<br></br>Edit         |                         View                          |             View<br></br>Add<br></br>Edit             |             View<br></br>Add<br></br>Edit             |
+|   Saved Groups    |       View        |             View              |         View<br></br>Add<br></br>Edit         |                         View                          |             View<br></br>Add<br></br>Edit             |             View<br></br>Add<br></br>Edit             |
+|       Tags        |        No         |              No               |         View<br></br>Add<br></br>Edit         |                         View                          |             View<br></br>Add<br></br>Edit             |             View<br></br>Add<br></br>Edit             |
+| Slack Integration |        No         |              No               |         View<br></br>Add<br></br>Edit         |                         View                          |             View<br></br>Add<br></br>Edit             |             View<br></br>Add<br></br>Edit             |
```
[the remark this answers is not in the document]
Comment (lukesonnet) [reply docs/docs/account/user-permissions.mdx:37]: IDK but it seems reasonable to have a role that lets them do anything that isn't explicit team/billing management

[the remark this answers is not in the document]
Comment (mknowlton89) [reply docs/docs/account/user-permissions.mdx:37]: This is a bug in the code - only admins can view / add / edit Slack Integrations - there is a bug in the code that is showing the Slack Integration link in the SideNav, but when clicked, the user get's a "You don't have permission" message. I will fix the bug in a separate PR, but the permission chart has been updated.

**Pull request merged**